search/search-client/src/main/java/io/camunda/search/clients/query/SearchIdsQuery.java

```diff
@@ -7,37 +7,30 @@
  */
 package io.camunda.search.clients.query;
 
-import static io.camunda.util.CollectionUtil.addValuesToList;
-import static io.camunda.util.CollectionUtil.collectValues;
-
 import io.camunda.util.ObjectBuilder;
-import java.util.Collections;
+import java.util.ArrayList;
+import java.util.Arrays;
 import java.util.List;
 import java.util.Objects;
-import java.util.function.Function;
-
-public final record SearchIdsQuery(List<String> values) implements SearchQueryOption {
 
-  static SearchIdsQuery of(final Function<Builder, ObjectBuilder<SearchIdsQuery>> fn) {
-    return SearchQueryBuilders.ids(fn);
-  }
+public record SearchIdsQuery(List<String> values) implements SearchQueryOption {
 
   public static final class Builder implements ObjectBuilder<SearchIdsQuery> {
 
-    private List<String> ids;
+    private final List<String> ids = new ArrayList<>();
 
     public Builder values(final List<String> values) {
-      ids = addValuesToList(ids, values);
+      ids.addAll(Objects.requireNonNullElse(values, List.of()));
```

Comment:
@Zelldon there will be a change of behaviour here
**before** 
`new Builder().values(List.of("a")).values(List.of("b"))` gives `ids` with `["b"]`
**now**
`new Builder().values(List.of("a")).values(List.of("b"))` gives `ids` with `["a", "b"]`

I think it is not expected to invoke many time `Builder.values` but I am wondering if this is a desired result 🤔 ?

Reply:
Hm 🤔 are you sure? I thought that was intended to have a combined list, this is how I understood the code of `addValuesToList` (and the name :D )